doc/decisions/map_undo_redo_implementation.md

```diff
@@ -0,0 +1,149 @@
+# Map Undo/Redo Implementation
+
+## Problem
+
+Undo/redo functionality is a common feature in many applications, PermaplanT is no exception.
+The user should be able to undo/redo the actions taken within the canvas.
+
+## Constraints
+
+1. The implementation of undo/redo should not have a significant impact on the performance of the app.
+2. The implementation of undo/redo should correspond to the related [use case](/doc/usecases/map_undo_redo.md)
+
+## Assumptions
+
+N/A
+
+## Solutions
+
+### Store every step of the user in the database
+
+The idea is to store every step of the user taken within the canvas in the database.
```

Comment:
indexedDB was considered in the offloading decision but was rejected for the current development phase. Based on your comments, I understood the confusion and applied changes to the offloading decision accordingly. Here, everything should point to the common term of the `database` as it is, otherwise, it will lead to a mess and confusion. Many thanks for pointing it out!